lib/fuzzer/mutate.zig

```diff
@@ -0,0 +1,339 @@
+        }
+    }
+}
+
+/// Writes names of the mutations that correspond to this seed. Used to make
+/// the log pretty
+pub fn writeMutation(seed: u64, writer: anytype) !void {
+    const muts = generateRandomMutationSequence(Rng.init(seed));
+    for (muts.slice(), 0..) |mut, i| {
+        if (i != 0) try writer.writeAll(", ");
+        try writer.writeAll(switch (mut) {
+            .shuffle_bytes => "Shuffle",
+            .erase_bytes => "DelBytes",
+            .insert_byte => "InsByte",
+            .insert_repeated_byte => "InsBytes",
+            .change_byte => "ChByte",
+            .change_bit => "ChBit",
+        });
+    }
+}
+
+fn mutateChangeBit(str: []u8, index: u32, bit: u3) void {
+    if (str.len == 0) return;
+    const mask = @as(u8, 1) << bit;
+    str[index % str.len] ^= mask;
+}
+
+fn mutateChangeBitReverse(str: []u8, index: u32, bit: u3) void {
+    return mutateChangeBit(str, index, bit);
+}
+
+test "mutate change bit" {
+    var scr = ArrayList(u8).init(std.testing.allocator);
+    var str = ArrayList(u8).init(std.testing.allocator);
+    defer str.deinit();
+    defer scr.deinit();
+    var rng = Rng.init(0);
+    for (0..test_data.len) |l| {
+        str.clearRetainingCapacity();
+        try str.appendSlice(test_data[0..l]);
+        for (0..1000) |_| {
+            const index: u32 = @truncate(rng.next());
+            const bit: u3 = @truncate(rng.next());
+            mutateChangeBit(&str, index, bit);
+            mutateChangeBitReverse(str.items, index, bit);
+            try std.testing.expectEqualStrings(test_data[0..l], str.items);
+            try std.testing.expectEqual(0, scr.items.len);
+        }
+    }
+}
+
+fn mutateChangeByte(str: []u8, scr: *ArrayList(u8), index: u32, byte: u8) !void {
+    if (str.len == 0) return;
+    const target = &str[index % str.len];
+    try scr.append(target.*);
+    target.* = byte;
+}
+
+fn mutateChangeByteReverse(str: []u8, scr: *ArrayListUnmanaged(u8), index: u32) void {
+    if (str.len == 0) return;
+    str[index % str.len] = scr.pop();
+}
+
+test "mutate change byte" {
+    var scr = ArrayList(u8).init(std.testing.allocator);
+    defer scr.deinit();
+    var str = ArrayList(u8).init(std.testing.allocator);
+    defer str.deinit();
+    var rng = Rng.init(0);
+    for (0..test_data.len) |l| {
+        str.clearRetainingCapacity();
+        try str.appendSlice(test_data[0..l]);
+        for (0..1000) |_| {
+            const index: u32 = @truncate(rng.next());
+            const byte: u8 = @truncate(rng.next());
+            try mutateChangeByte(str.items, &scr, index, byte);
+            mutateChangeByteReverse(&str, &scr, index);
+            try std.testing.expectEqualStrings(test_data[0..l], str.items);
+            try std.testing.expectEqual(0, scr.items.len);
+        }
+    }
+}
+
+fn mutateInsertRepeatedByte(str: *ArrayList(u8), index: u32, len_: u8, byte: u8) !void {
+    const len = @min(24, @max(1, len_));
```

Comment:
This is very biased (I don't know if you want it to be.) You should assert that it is within the range and generate an unbiased number at the call site.